gateway/api_definition.go

```diff
@@ -285,7 +285,7 @@ type APIDefinitionLoader struct {
 
 // MakeSpec will generate a flattened URLSpec from and APIDefinitions' VersionInfo data. paths are
 // keyed to the Api version name, which is determined during routing to speed up lookups
-func (a APIDefinitionLoader) MakeSpec(def *apidef.APIDefinition, logger *logrus.Entry) *APISpec {
+func (a APIDefinitionLoader) MakeSpec(def nestedApiDefinition, logger *logrus.Entry) *APISpec {
```

Comment:
We should use a pointer receiver here.

- lower memory pressure (less allocations)
- interfaces are never *io.Reader, so while reading this line it would be immediately clear that *nestedApiDefinition is an *struct, while without it it may be an interface you need to look up. Adding the * improves understanding/readability when reviewing code.
- the struct itself contains pointers, passing it by value may imply that it's a copy, but a copy of a pointer inside it is modifiable. Passing it by reference makes it clear you may be doing something undesired if/when you modify the data in the value.